### PR TITLE
Enable gci, goprintffuncname, nolintlint linters

### PR DESCRIPTION
**Pull request by ankitjain235**
## Change Overview

This PR enables gci linter to group imports.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [x] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- #2173
- #2568

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: cmd/reposervercontroller/main.go [hunk 1]
```diff
@@ -38,7 +38,6 @@ import (
 	"github.com/kanisterio/kanister/pkg/log"
 	"github.com/kanisterio/kanister/pkg/resource"
 	"github.com/kanisterio/kanister/pkg/validatingwebhook"
-	//+kubebuilder:scaffold:imports
```
Comment (PrasadG193) [inline cmd/reposervercontroller/main.go:41]: Its a marker, not import. I tried to find what exactly it does, bit couldn't find much info.
Is it safe to remove this?

File: pkg/client/informers/externalversions/cr/v1alpha1/repositoryserver.go [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2023 The Kanister Authors.
+Copyright The Kubernetes Authors.
```
Comment (PrasadG193) [inline pkg/client/informers/externalversions/cr/v1alpha1/repositoryserver.go:2]: Unintended changes?

File: pkg/client/listers/cr/v1alpha1/repositoryserver.go [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
-Copyright 2023 The Kanister Authors.
+Copyright The Kubernetes Authors.
```
Comment (PrasadG193) [inline pkg/client/listers/cr/v1alpha1/repositoryserver.go:2]: same here

File: cmd/reposervercontroller/main.go (hunk 1 above)
Comment (ankitjain235) [reply cmd/reposervercontroller/main.go:41]: I think it got removed automatically. Will re-add it until we know for sure it's not needed.

Comment (saima-s) [reply cmd/reposervercontroller/main.go:43]: Added it back. This [issue](https://github.com/kubernetes-sigs/kubebuilder/issues/1487#issuecomment-823864494) explains the reason of this. We can focus on this [part](https://github.com/kubernetes-sigs/kubebuilder/issues/1487#issuecomment-1485346413) of it.

File: pkg/client/informers/externalversions/cr/v1alpha1/repositoryserver.go (hunk 2 above)
Comment (saima-s) [reply pkg/client/informers/externalversions/cr/v1alpha1/repositoryserver.go]: reverted as before.

File: pkg/client/listers/cr/v1alpha1/repositoryserver.go (hunk 3 above)
Comment (saima-s) [reply pkg/client/listers/cr/v1alpha1/repositoryserver.go]: reverted

File: pkg/kube/exec_test.go [hunk 4]
```diff
@@ -274,7 +274,7 @@ func (s *ExecSuite) TestKopiaCommand(c *C) {
 	}
 	p, err := s.cli.CoreV1().Pods(s.namespace).Create(ctx, pod, metav1.CreateOptions{})
 	c.Assert(err, IsNil)
-	defer s.cli.CoreV1().Pods(s.namespace).Delete(ctx, pod.Name, metav1.DeleteOptions{}) //nolint: errcheck
+	defer s.cli.CoreV1().Pods(s.namespace).Delete(ctx, pod.Name, metav1.DeleteOptions{})
```
[the remark this answers is not in the document]
Comment (saima-s) [reply pkg/kube/exec_test.go:277]: test is excluded from linters.

File: pkg/client/listers/cr/v1alpha1/repositoryserver.go [hunk 5]
```diff
@@ -20,6 +20,7 @@ package v1alpha1
 
 import (
 	v1alpha1 "github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+
```
[the remark this answers is not in the document]
Comment (saima-s) [reply pkg/client/listers/cr/v1alpha1/repositoryserver.go:23]: This is generated file. Regenrated it in a different PR.

File: .golangci.yml [hunk 6]
```diff
@@ -1,5 +1,5 @@
 linters:
-  disable-all: false                    # This enabled 7 default linters as of golangci-lint@v1.50.0:
+  disable-all: true                    # This enabled 7 default linters as of golangci-lint@v1.50.0:
```
Comment (ankitjain235) [inline .golangci.yml:2]: Isn't it disabling the default linters? Why do we need this change?

Comment (ankitjain235) [reply .golangci.yml:2]: https://golangci-lint.run/usage/linters/#enabled-by-default

File: pkg/blockstorage/azure/azuredisk.go [hunk 7]
```diff
@@ -1,8 +1,5 @@
 // TODO: Switch to using the latest azure sdk and remove nolint.
 // Related Ticket- https://github.com/kanisterio/kanister/issues/1684
-//
-//nolint:staticcheck
-
```
Comment (ankitjain235) [inline pkg/blockstorage/azure/azuredisk.go:5]: I believe this is the result of https://github.com/kanisterio/kanister/pull/2500/files#diff-6179837f7df53a6f05c522b6b7bb566d484d5465d9894fb04910dd08bb40dcc9R2

**Review by ankitjain235 — COMMENTED** [review]
Just one question around why are we disabling default linters.

File: .golangci.yml (hunk 6 above)
Comment (saima-s) [reply .golangci.yml:2]: yes fixing this.

File: pkg/blockstorage/azure/azuredisk.go (hunk 7 above)
Comment (saima-s) [reply pkg/blockstorage/azure/azuredisk.go:5]: done

File: cmd/reposervercontroller/main.go [hunk 8]
```diff
@@ -40,7 +39,10 @@ import (
 	"github.com/kanisterio/kanister/pkg/log"
 	"github.com/kanisterio/kanister/pkg/resource"
 	"github.com/kanisterio/kanister/pkg/validatingwebhook"
+	//nolint:gci
```
Comment (viveksinghggits) [inline cmd/reposervercontroller/main.go:42]: what is the error that we get if we don't ignore this here?

File: pkg/blueprint/validate/validate.go [hunk 9]
```diff
@@ -21,9 +21,10 @@ import (
 
 	kanister "github.com/kanisterio/kanister/pkg"
 	crv1alpha1 "github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
-	_ "github.com/kanisterio/kanister/pkg/function"
 	"github.com/kanisterio/kanister/pkg/param"
 	"github.com/kanisterio/kanister/pkg/utils"
+
+	_ "github.com/kanisterio/kanister/pkg/function"
```
Comment (viveksinghggits) [inline pkg/blueprint/validate/validate.go:27]: does this need to be grouped separately, it's also a kanister dep right?

**Review by viveksinghggits — COMMENTED** [review]
LGTM

File: pkg/blueprint/validate/validate.go (hunk 9 above)
Comment (saima-s) [reply pkg/blueprint/validate/validate.go:27]: yes, it is grouped as blank in gci linter.

File: cmd/reposervercontroller/main.go (hunk 8 above)
Comment (saima-s) [reply cmd/reposervercontroller/main.go:42]: when I run `golangci-lint run` it automatically removes `//+kubebuilder:scaffold:imports` which we don't want to.

**Review by viveksinghggits — APPROVED** [review]

**Pull request merged**